tensorly/regression/cp_plsr.py

```diff
@@ -256,11 +301,11 @@ def transform(self, X, Y=None):
 
                 Y -= T.dot(
                     T.dot(
-                        T.lstsq(T.transpose(X_scores), T.transpose(X_scores))[0],
-                        Y_scores[:, [component]],
+                        X_scores,
+                        T.reshape(self.coef_[:, component], (-1, 1)),
                     ),
-                    T.transpose(self.Y_factors[1][:, [component]]),
-                )  # Y -= T pinv(T) u q'
+                    T.transpose(T.reshape(self.Y_factors[1][:, component], (-1, 1))),
```

Comment:
Same, do we need the reshape? It's just a vector.

Reply:
Same, but `T.transpose(T.reshape)` is redundant indeed.